lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -182,6 +182,30 @@ def get_cpu_facts(self, collected_facts=None):
             return cpu_facts
 
         cpu_facts['processor'] = []
+        cmd = self.module.get_bin_path('lscpu')
```

Comment:
Use `get_bin_path()` from `ansible.module_utils.common.process`, make this command required, then catch the `ValueError` returned when the command is not present.
```suggestion
        try:
            cmd = get_bin_path('lscpu', required=True)
            # existing code than returns cpu_facts
        except ValueError as e:
            # existing code to process `/proc/cpuinfo`
```

Reply:
fixed